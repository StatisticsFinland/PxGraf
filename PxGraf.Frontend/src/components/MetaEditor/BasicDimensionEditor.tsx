import React from 'react';
import { useTranslation } from 'react-i18next';
import { Stack } from '@mui/material';
import { UiLanguageContext } from 'contexts/uiLanguageContext';
import { EditorField } from './Editorfield';
import styled from 'styled-components';
import { IDimension } from 'types/cubeMeta';
import { IDimensionEditions } from 'types/query';

const EditorFieldWrapper = styled(Stack)`
  padding: 0px;
`;

interface IBasicDimensionEditor {
    dimension: IDimension;
    language: string;
    dimensionEdits: IDimensionEditions;
    onChange: (newEdit: IDimensionEditions) => void;
}

export const BasicDimensionEditor: React.FC<IBasicDimensionEditor> = ({ dimension, language, dimensionEdits, onChange }) => {
    const { t } = useTranslation();
    const { uiContentLanguage } = React.useContext(UiLanguageContext);

    return (
        <EditorFieldWrapper spacing={2}>
            {dimension.values.map(value => {
                return (
                    <EditorField
                        label={t("editMetadata.valueName") + ": " + value.name[uiContentLanguage]}
                        key={value.code}
                        defaultValue={value.name[language]}
                        editValue={dimensionEdits?.valueEdits[value.code]?.nameEdit?.[language]}
                        onChange={newValue => {
                            const newDimensionEdit = {
                                ...dimensionEdits,
                                valueEdits: {
                                    ...dimensionEdits?.valueEdits,
                                    [value.code]: {
                                        ...dimensionEdits?.valueEdits[value.code],
                                        nameEdit: {
                                            ...dimensionEdits?.valueEdits[value.code]?.nameEdit,
                                            [language]: newValue
                                        }
                                    }
                                }
                            };
                            onChange(newDimensionEdit);
                        }}
                    />
                );
            })}
        </EditorFieldWrapper>
    );
}

export default BasicDimensionEditor;