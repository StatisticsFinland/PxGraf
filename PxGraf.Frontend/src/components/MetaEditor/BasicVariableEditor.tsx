import React from 'react';
import { useTranslation } from 'react-i18next';
import { Stack } from '@mui/material';
import { UiLanguageContext } from 'contexts/uiLanguageContext';
import { EditorField } from './Editorfield';
import styled from 'styled-components';
import { IDimension } from 'types/cubeMeta';
import { IVariableEditions } from 'types/query';

const EditorFieldWrapper = styled(Stack)`
  padding: 0px;
`;

interface IBasicVariableEditorProps {
    variable: IDimension;
    language: string;
    variableEdits: IVariableEditions;
    onChange: (newEdit: IVariableEditions) => void;
}

export const BasicVariableEditor: React.FC<IBasicVariableEditorProps> = ({ variable, language, variableEdits, onChange }) => {
    const { t } = useTranslation();
    const { uiContentLanguage } = React.useContext(UiLanguageContext);

    return (
        <EditorFieldWrapper spacing={2}>
            {variable.Values.map(value => {
                return (
                    <EditorField
                        label={t("editMetadata.valueName") + ": " + value.Name[uiContentLanguage]}
                        key={value.Code}
                        defaultValue={value.Name[language]}
                        editValue={variableEdits?.valueEdits[value.Code]?.nameEdit?.[language]}
                        onChange={newValue => {
                            const newVariableEdit = {
                                ...variableEdits,
                                valueEdits: {
                                    ...variableEdits?.valueEdits,
                                    [value.Code]: {
                                        ...variableEdits?.valueEdits[value.Code],
                                        nameEdit: {
                                            ...variableEdits?.valueEdits[value.Code]?.nameEdit,
                                            [language]: newValue
                                        }
                                    }
                                }
                            };
                            onChange(newVariableEdit);
                        }}
                    />
                );
            })}
        </EditorFieldWrapper>
    );
}

export default BasicVariableEditor;