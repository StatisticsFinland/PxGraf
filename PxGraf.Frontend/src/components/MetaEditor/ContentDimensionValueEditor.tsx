import React from 'react';
import { useTranslation } from 'react-i18next';
import { Stack, Paper } from '@mui/material';
import { UiLanguageContext } from 'contexts/uiLanguageContext';
import EditorField from './Editorfield';
import styled from 'styled-components';
import { IContentDimensionValue } from 'types/cubeMeta';
import { IDimensionValueEditions } from 'types/query';
import { getAdditionalPropertyValue } from '../../utils/metadataUtils';
import { sourceKey } from '../../utils/keywordConstants';

const EditorFieldWrapper = styled(Stack)`
  padding: 16px;
`;

interface IContentDimensionValueEditorProps {
    dimensionValue: IContentDimensionValue;
    language: string;
    valueEdits: IDimensionValueEditions;
    onChange: (newEdit: IDimensionValueEditions) => void;
}

export const ContentDimensionValueEditor: React.FC<IContentDimensionValueEditorProps> = ({ dimensionValue, language, valueEdits, onChange }) => {
    const { uiContentLanguage } = React.useContext(UiLanguageContext);

    const { t } = useTranslation();

    return (
        <Paper variant="outlined">
            <EditorFieldWrapper spacing={2}>
                <EditorField
                    label={t("editMetadata.valueName") + ": " + dimensionValue.name[uiContentLanguage]}
                    defaultValue={dimensionValue.name[language]}
                    editValue={valueEdits?.nameEdit?.[language]}
                    onChange={newValue => {
                        const newValueEdit: IDimensionValueEditions = {
                            ...valueEdits,
                            nameEdit: {
                                ...valueEdits?.nameEdit,
                                [language]: newValue
                            }
                        };
                        onChange(newValueEdit);
                    }}
                />
                <EditorField
                    label={t("editMetadata.unit")}
                    defaultValue={dimensionValue?.unit[language] ?? ''}
                    editValue={valueEdits?.contentComponent?.unitEdit?.[language]}
                    onChange={newValue => {
                        const newValueEdit: IDimensionValueEditions = {
                            ...valueEdits,
                            contentComponent: {
                                ...valueEdits?.contentComponent,
                                unitEdit: {
                                    ...valueEdits?.contentComponent?.unitEdit,
                                    [language]: newValue
                                }
                            }
                        };
                        onChange(newValueEdit);
                    }}
                />
                <EditorField
                    label={t("editMetadata.source")}
                    defaultValue={getAdditionalPropertyValue(sourceKey, dimensionValue?.additionalProperties)[language] ?? ''}
                    editValue={valueEdits?.contentComponent?.sourceEdit?.[language]}
                    onChange={newValue => {
                        const newValueEdit: IDimensionValueEditions = {
                            ...valueEdits,
                            contentComponent: {
                                ...valueEdits?.contentComponent,
                                sourceEdit: {
                                    ...valueEdits?.contentComponent?.sourceEdit,
                                    [language]: newValue
                                }
                            }
                        };
                        onChange(newValueEdit);
                    }}
                />
            </EditorFieldWrapper>
        </Paper>
    );
}

export default ContentDimensionValueEditor;