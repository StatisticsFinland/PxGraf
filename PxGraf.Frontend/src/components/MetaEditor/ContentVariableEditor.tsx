import { Grid } from '@mui/material';
import React from 'react';
import { IVariable } from 'types/cubeMeta';
import { IVariableEditions, IVariableValueEditions } from 'types/query';
import { ContentVariableValueEditor } from './ContentVariableValueEditor';

interface IContentVariableEditorProps {
    variable: IVariable;
    language: string;
    variableEdits: IVariableEditions;
    onChange: (newEdit: IVariableEditions) => void;
}

export const ContentVariableEditor: React.FC<IContentVariableEditorProps> = ({ variable, language, variableEdits, onChange }) => {

    const handleChange = (newValueEdits: IVariableValueEditions, code: string) => {
        const newVariableEdit = {
            ...variableEdits,
            valueEdits: {
                ...variableEdits?.valueEdits,
                [code]: newValueEdits
            }
        };

        onChange(newVariableEdit);
    };

    return (
        <Grid container spacing={3}>
            {variable.values.map((value) => {
                const valueEdits = variableEdits?.valueEdits[value.code];

                return (
                    <Grid item
                        key={value.code}
                        xs={12}
                        xl={variable.values.length > 1 ? 6 : 12 /* Use two columns layout when screen is big enought and there is more than one value */}
                    >
                        <ContentVariableValueEditor
                            key={value.code}
                            variableValue={value}
                            language={language}
                            valueEdits={valueEdits}
                            onChange={(newEdit) => handleChange(newEdit, value.code)}
                        />
                    </Grid>
                );
            })}
        </Grid>
    );
}