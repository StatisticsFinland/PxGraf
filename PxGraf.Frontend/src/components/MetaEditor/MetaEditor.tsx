import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Box, Typography, Accordion, AccordionSummary, AccordionDetails, Tabs, Tab } from '@mui/material';
import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
import { TabPanel } from 'components/TabPanel/TabPanel';
import { DimensionEditor } from './DimensionEditor';
import { HeaderEditor } from './HeaderEditor';
import { a11yProps } from 'utils/componentHelpers';
import styled from 'styled-components';
import { IDimension } from 'types/cubeMeta';
import InfoBubble from 'components/InfoBubble/InfoBubble';
import { UiLanguageContext } from 'contexts/uiLanguageContext';
import { IEditorContentsResult } from '../../api/services/editor-contents';

const MetaEditorWrapper = styled(Box)`
  grid-area: 'parameters';
  display: grid;
  gap: 16px;
  grid-template-columns: 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr 1fr;
  padding: 0px;
`;

const StyledAccordion = styled(Accordion)`
  width: 100%;
`;

const TabsWrapper = styled(Box)`
  border-bottom: 1px solid;
  grid-column: span 12;
`;

const TabPanelWrapper = styled(Box)`
  grid-column: span 12;
`;

const MetaEditBarWrapper = styled.div`
  display: flex;
`;

const GridFixer = styled.div`
  grid-column: span 12;
`;

const StyledAccordionDetails = styled(AccordionDetails)`
  background-color: #f8f8f8;
`;

export interface INewEditMetaEditor {
    dimensionQueries?: {
        [dimensionCode: string]: string;
    };
    chartHeaderEdit?: {
        [language: string]: string;
    };
}

interface IMetaEditorProps {
    language: string;
    resolvedDimensions: IDimension[];
    editorContentsResponse: IEditorContentsResult;
    isMetaAccordionOpen: boolean;
    onMetaAccordionOpenChange: () => void;
    titleMaxLength?: number;
}


export const MetaEditor: React.FC<IMetaEditorProps> = ({
    language,
    resolvedDimensions,
    editorContentsResponse,
    isMetaAccordionOpen,
    onMetaAccordionOpenChange,
    titleMaxLength,
}) => {
    const { t } = useTranslation();
    const [dimensionTab, setDimensionTab] = useState(0);
    const { uiContentLanguage } = React.useContext(UiLanguageContext);

    return (
        <MetaEditorWrapper>
            <HeaderEditor
                style={{ width: '100%' }}
                editorContentResponse={editorContentsResponse}
                language={language}
                maxLength={titleMaxLength}
            />
            <GridFixer>

                <MetaEditBarWrapper>

                    <InfoBubble info={t('infoText.metaEdition')} ariaLabel={t("editMetadata.editorTitle")} />
                    <StyledAccordion expanded={isMetaAccordionOpen} onChange={onMetaAccordionOpenChange}>
                        <AccordionSummary
                            style={{ width: '100%' }}
                            expandIcon={<ExpandMoreIcon />}
                            aria-controls="panel1a-content"
                            id="panel1a-header"
                        >
                            <Typography>{t("editMetadata.editorTitle")}</Typography>
                        </AccordionSummary>
                        <StyledAccordionDetails>
                            <TabsWrapper sx={{ borderColor: 'divider' }}>
                                <Tabs value={dimensionTab} onChange={(evt, newTab) => setDimensionTab(newTab)}>
                                    {resolvedDimensions.map((dimension, index) => <Tab label={dimension.name[uiContentLanguage] ?? dimension.code} {...a11yProps(index)} key={dimension.code} />)}
                                </Tabs>
                            </TabsWrapper>
                            <TabPanelWrapper>
                                {resolvedDimensions.map((dimension, index) => {
                                    return (
                                        <TabPanel selectedValue={dimensionTab} value={index} key={dimension.code}>
                                            <DimensionEditor
                                                dimension={dimension}
                                                language={language}
                                            />
                                        </TabPanel>
                                    );
                                })}
                            </TabPanelWrapper>
                        </StyledAccordionDetails>
                    </StyledAccordion>
                </MetaEditBarWrapper>
            </GridFixer>
        </MetaEditorWrapper>
    );
};

export default MetaEditor;