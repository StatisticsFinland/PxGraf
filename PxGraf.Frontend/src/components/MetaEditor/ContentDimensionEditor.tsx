import { Grid } from '@mui/material';
import React from 'react';
import { IDimension } from 'types/cubeMeta';
import { IDimensionEditions, IDimensionValueEditions } from 'types/query';
import { ContentDimensionValueEditor } from './ContentDimensionValueEditor';

interface IContentDimensionEditorProps {
    dimension: IDimension;
    language: string;
    dimensionEdits: IDimensionEditions;
    onChange: (newEdit: IDimensionEditions) => void;
}

export const ContentDimensionEditor: React.FC<IContentDimensionEditorProps> = ({ dimension, language, dimensionEdits, onChange }) => {

    const handleChange = (newValueEdits: IDimensionValueEditions, code: string) => {
        const newDimensionEdit = {
            ...dimensionEdits,
            valueEdits: {
                ...dimensionEdits?.valueEdits,
                [code]: newValueEdits
            }
        };

        onChange(newDimensionEdit);
    };

    return (
        <Grid container spacing={3}>
            {dimension.values.map((value) => {
                const valueEdits = dimensionEdits?.valueEdits[value.code];

                return (
                    <Grid item
                        key={value.code}
                        xs={12}
                        xl={dimension.values.length > 1 ? 6 : 12 /* Use two columns layout when screen is big enought and there is more than one value */}
                    >
                        <ContentDimensionValueEditor
                            key={value.code}
                            dimensionValue={value}
                            language={language}
                            valueEdits={valueEdits}
                            onChange={(newEdit) => handleChange(newEdit, value.code)}
                        />
                    </Grid>
                );
            })}
        </Grid>
    );
}