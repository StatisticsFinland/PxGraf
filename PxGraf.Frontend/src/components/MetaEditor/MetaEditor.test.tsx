import { render } from '@testing-library/react';
import { IHeaderResult } from 'api/services/default-header';
import React from 'react';
import { IVariable, VariableType } from 'types/cubeMeta';
import { ICubeQuery } from 'types/query';
import MetaEditor from './MetaEditor';
import UiLanguageContext from 'contexts/uiLanguageContext';

jest.mock('react-i18next', () => ({
    ...jest.requireActual('react-i18next'),
    useTranslation: () => {
        return {
            t: (str: string) => str,
            i18n: {
                changeLanguage: () => new Promise(() => null),
            },
        };
    },
}));

const setLanguage = jest.fn();
const language = 'fi';
const setLanguageTab = jest.fn();
const languageTab = 'fi';
const availableUiLanguages = ['fi', 'en', 'sv'];
const uiContentLanguage = 'fi';
const setUiContentLanguage = jest.fn();

const mockVariables: IVariable[] = [{
        code: 'foo',
        name: {
            'fi': 'asd',
            'sv': 'asd',
            'en': 'asd'
        },
        note: {
            'fi': 'seppo',
            'sv': 'seppo',
            'en': 'seppo'
        },
        type: VariableType.Content,
        values: [
            {
                code: 'bar',
                isSum: false,
                name: {
                    'fi': 'fgfgfg',
                    'sv': 'fgfgfg',
                    'en': 'fgfgfg'
                },
                note: {
                    'fi': 'fghjfgh',
                    'sv': 'fghjfgh',
                    'en': 'fghjfgh'
                }
            }
        ]
    },
    {
        code: 'foo2',
        name: {
            'fi': null,
            'sv': null,
            'en': null
        },
        note: {
            'fi': 'seppo2',
            'sv': 'seppo2',
            'en': 'seppo2'
        },
        type: VariableType.Content,
        values: [
            {
                code: 'bar2',
                isSum: false,
                name: {
                    'fi': null,
                    'sv': null,
                    'en': null,
                },
                note: null
            }
            ]
    }
]
const mockLang = 'fi';
const mockFunction = jest.fn();

const isMetaAccordionOpenMock = true;

const defaultHeaderResponseMock: IHeaderResult = {
    isLoading: false,
    isError: false,
    data: {
        'fi': 'asd',
        'en': 'asd',
        'sv': 'asd'
    }
}

const cubeQueryMock: ICubeQuery = {
    chartHeaderEdit: {
        'fi': 'foo',
        'sv': 'foo',
        'en': 'foo'
    },
    variableQueries: {
        'asd123query': {
            valueEdits: {
                'code1': {
                    contentComponent: {
                        sourceEdit: {
                            'fi': '123',
                            'sv': '123',
                            'en': '123'
                        },
                        unitEdit: {
                            'fi': '456',
                            'sv': '456',
                            'en': '456'
                        }
                    }
                }
            }
        }
    }
};

describe('Rendering test', () => {
    it('renders correctly', () => {
        const { asFragment } = render(
            <UiLanguageContext.Provider value={{ language, setLanguage, languageTab, setLanguageTab, availableUiLanguages, uiContentLanguage, setUiContentLanguage }}>
                <MetaEditor resolvedVariables={mockVariables} cubeQuery={cubeQueryMock} defaultHeaderResponse={defaultHeaderResponseMock} isMetaAccordionOpen={isMetaAccordionOpenMock} language={mockLang} onChange={mockFunction} onMetaAccordionOpenChange={mockFunction} />
            </UiLanguageContext.Provider>
        );
        expect(asFragment()).toMatchSnapshot();
    });
});
