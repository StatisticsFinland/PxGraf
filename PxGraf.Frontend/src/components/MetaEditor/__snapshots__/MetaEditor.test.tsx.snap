// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering test renders correctly 1`] = `
<DocumentFragment>
  <div
    class="sc-keTIit iEZRuv MuiBox-root css-0"
  >
    <div
      class="sc-dstKZu beBTtl"
    >
      <div
        class="sc-khLCKb gYRtPq"
      >
        <button
          aria-label="tooltip.open: editMetadata.header"
          class="sc-ghWlax bEVwpO"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
            />
          </svg>
        </button>
        <div
          class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
          style="width: 100%;"
        >
          <label
            class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
            data-shrink="true"
            for=":r0:"
          >
            <b>
              editMetadata.header*
            </b>
          </label>
          <div
            class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd sc-gtLWhw bPTTGW edited css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
          >
            <input
              aria-invalid="false"
              class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
              id=":r0:"
              type="text"
              value="foo"
            />
            <div
              class="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
            >
              <span
                aria-label="editMetadata.discardChanges"
                class=""
                data-mui-internal-clone-element="true"
              >
                <button
                  aria-label="editMetadata.discardChanges"
                  class="MuiButtonBase-root MuiIconButton-root MuiIconButton-edgeEnd MuiIconButton-sizeMedium css-1yq5fb3-MuiButtonBase-root-MuiIconButton-root"
                  tabindex="0"
                  type="button"
                >
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                    data-testid="UndoIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M12.5 8c-2.65 0-5.05.99-6.9 2.6L2 7v9h9l-3.62-3.62c1.39-1.16 3.16-1.88 5.12-1.88 3.54 0 6.55 2.31 7.6 5.5l2.37-.78C21.08 11.03 17.15 8 12.5 8"
                    />
                  </svg>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
              </span>
            </div>
            <fieldset
              aria-hidden="true"
              class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
            >
              <legend
                class="css-14lo706"
              >
                <span>
                  editMetadata.header
                </span>
              </legend>
            </fieldset>
          </div>
          <div
            aria-live="polite"
          />
        </div>
      </div>
    </div>
    <div
      class="sc-cEzcPc iUUCmm"
    >
      <div
        class="sc-jtQUzJ drtSIt"
      >
        <button
          aria-label="tooltip.open: editMetadata.editorTitle"
          class="sc-ghWlax bEVwpO"
        >
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSvgIcon-colorInfo MuiSvgIcon-fontSizeMedium css-1yq1soz-MuiSvgIcon-root"
            data-testid="InfoIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2m1 15h-2v-6h2zm0-8h-2V7h2z"
            />
          </svg>
        </button>
        <div
          class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiAccordion-root MuiAccordion-rounded Mui-expanded MuiAccordion-gutters sc-dpBQxM dvJsbi css-1086bdv-MuiPaper-root-MuiAccordion-root"
        >
          <div
            aria-controls="panel1a-content"
            aria-expanded="true"
            class="MuiButtonBase-root MuiAccordionSummary-root Mui-expanded MuiAccordionSummary-gutters css-1f773le-MuiButtonBase-root-MuiAccordionSummary-root"
            id="panel1a-header"
            role="button"
            style="width: 100%;"
            tabindex="0"
          >
            <div
              class="MuiAccordionSummary-content Mui-expanded MuiAccordionSummary-contentGutters css-eqpfi5-MuiAccordionSummary-content"
            >
              <p
                class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
              >
                editMetadata.editorTitle
              </p>
            </div>
            <div
              class="MuiAccordionSummary-expandIconWrapper Mui-expanded css-yw020d-MuiAccordionSummary-expandIconWrapper"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                data-testid="ExpandMoreIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M16.59 8.59 12 13.17 7.41 8.59 6 10l6 6 6-6z"
                />
              </svg>
            </div>
          </div>
          <div
            class="MuiCollapse-root MuiCollapse-vertical MuiCollapse-entered css-pwcg7p-MuiCollapse-root"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper MuiCollapse-vertical css-smkl36-MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner MuiCollapse-vertical css-9l5vo-MuiCollapse-wrapperInner"
              >
                <div
                  aria-labelledby="panel1a-header"
                  class="MuiAccordion-region"
                  id="panel1a-content"
                  role="region"
                >
                  <div
                    class="MuiAccordionDetails-root sc-ovuCP kmuGeq css-15v22id-MuiAccordionDetails-root"
                  >
                    <div
                      class="sc-jwIPbr itDDef MuiBox-root css-1mc6bnb"
                    >
                      <div
                        class="MuiTabs-root css-1ujnqem-MuiTabs-root"
                      >
                        <div
                          class="MuiTabs-scroller MuiTabs-fixed css-jpln7h-MuiTabs-scroller"
                          style="overflow: hidden; margin-bottom: 0px;"
                        >
                          <div
                            class="MuiTabs-flexContainer css-heg063-MuiTabs-flexContainer"
                            role="tablist"
                          >
                            <button
                              aria-controls="simple-tabpanel-0"
                              aria-selected="true"
                              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary Mui-selected css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                              id="simple-tab-0"
                              role="tab"
                              tabindex="0"
                              type="button"
                            >
                              asd
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                            <button
                              aria-controls="simple-tabpanel-1"
                              aria-selected="false"
                              class="MuiButtonBase-root MuiTab-root MuiTab-textColorPrimary css-1h9z7r5-MuiButtonBase-root-MuiTab-root"
                              id="simple-tab-1"
                              role="tab"
                              tabindex="-1"
                              type="button"
                            >
                              foo2
                              <span
                                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                              />
                            </button>
                          </div>
                          <span
                            class="MuiTabs-indicator css-1aquho2-MuiTabs-indicator"
                            style="left: 0px; width: 0px;"
                          />
                        </div>
                      </div>
                    </div>
                    <div
                      class="sc-cHqXqK FSQTe MuiBox-root css-0"
                    >
                      <div
                        aria-labelledby="simple-tab-0"
                        id="simple-tabpanel-0"
                        role="tabpanel"
                      >
                        <div
                          class="sc-blHHSb dKmohW MuiBox-root css-0"
                        >
                          <div
                            class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-3 css-zow5z4-MuiGrid-root"
                          >
                            <div
                              class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-xl-12 css-1idn90j-MuiGrid-root"
                            >
                              <div
                                class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded css-172kgz0-MuiPaper-root"
                              >
                                <div
                                  class="MuiStack-root sc-egkSDF emnLII css-1sazv7p-MuiStack-root"
                                >
                                  <div
                                    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                                  >
                                    <label
                                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                                      data-shrink="true"
                                      for=":r2:"
                                    >
                                      editMetadata.valueName: fgfgfg
                                    </label>
                                    <div
                                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl sc-gtLWhw bPTTGW default css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                                        id=":r2:"
                                        type="text"
                                        value="fgfgfg"
                                      />
                                      <fieldset
                                        aria-hidden="true"
                                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                      >
                                        <legend
                                          class="css-14lo706"
                                        >
                                          <span>
                                            editMetadata.valueName: fgfgfg
                                          </span>
                                        </legend>
                                      </fieldset>
                                    </div>
                                    <div
                                      aria-live="polite"
                                    />
                                  </div>
                                  <div
                                    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                                  >
                                    <label
                                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                                      data-shrink="true"
                                      for=":r3:"
                                    >
                                      editMetadata.unit
                                    </label>
                                    <div
                                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl sc-gtLWhw bPTTGW default css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                                        id=":r3:"
                                        type="text"
                                        value="yksikko"
                                      />
                                      <fieldset
                                        aria-hidden="true"
                                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                      >
                                        <legend
                                          class="css-14lo706"
                                        >
                                          <span>
                                            editMetadata.unit
                                          </span>
                                        </legend>
                                      </fieldset>
                                    </div>
                                    <div
                                      aria-live="polite"
                                    />
                                  </div>
                                  <div
                                    class="MuiFormControl-root css-1nrlq1o-MuiFormControl-root"
                                  >
                                    <label
                                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiFormLabel-filled MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-sizeMedium MuiInputLabel-outlined css-1jy569b-MuiFormLabel-root-MuiInputLabel-root"
                                      data-shrink="true"
                                      for=":r4:"
                                    >
                                      editMetadata.source
                                    </label>
                                    <div
                                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl sc-gtLWhw bPTTGW default css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                                    >
                                      <input
                                        aria-invalid="false"
                                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                                        id=":r4:"
                                        type="text"
                                        value="lahde"
                                      />
                                      <fieldset
                                        aria-hidden="true"
                                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                                      >
                                        <legend
                                          class="css-14lo706"
                                        >
                                          <span>
                                            editMetadata.source
                                          </span>
                                        </legend>
                                      </fieldset>
                                    </div>
                                    <div
                                      aria-live="polite"
                                    />
                                  </div>
                                </div>
                              </div>
                            </div>
                          </div>
                        </div>
                      </div>
                      <div
                        aria-labelledby="simple-tab-1"
                        hidden=""
                        id="simple-tabpanel-1"
                        role="tabpanel"
                      />
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
